apiserver/facades/client/bundle/bundle.go

```diff
@@ -248,7 +248,44 @@ func (b *BundleAPI) fillBundleData(model description.Model) (*charm.BundleData,
 			Annotations: machine.Annotations(),
 			Series:      machine.Series(),
 		}
-		if result := b.constraints(machine.Constraints()); len(result) != 0 {
+
+		result := b.constraints(machine.Constraints())
+
+		// Get the machine, so we can get the hardware characteristics.
+		// This is done for parity with GUI.
+		if machineState, _ := b.backend.Machine(machine.Id()); machineState != nil {
+			hardware, err := machineState.HardwareCharacteristics()
+			if err == nil {
+				result = result[:0]
```

Comment:
This logic isn't right.
We need to combine the machine constraints and h/w info before splitting the machine constraints into a string.
There's no tests for this new work which would have shown the problem.